src-self-hosted/Module.zig

```diff
@@ -732,17 +732,19 @@ pub const InitOptions = struct {
     object_format: ?std.builtin.ObjectFormat = null,
     optimize_mode: std.builtin.Mode = .Debug,
     keep_source_files_loaded: bool = false,
+    cbe: bool = false,
 };
 
 pub fn init(gpa: *Allocator, options: InitOptions) !Module {
     const bin_file_dir = options.bin_file_dir orelse std.fs.cwd();
-    var bin_file = try link.openBinFilePath(gpa, bin_file_dir, options.bin_file_path, .{
+    const bin_file = try link.openBinFilePath(gpa, bin_file_dir, options.bin_file_path, .{
         .target = options.target,
         .output_mode = options.output_mode,
         .link_mode = options.link_mode orelse .Static,
         .object_format = options.object_format orelse options.target.getObjectFormat(),
+        .cbe = options.cbe,
     });
-    errdefer bin_file.deinit();
+    errdefer bin_file.*.deinit();
```

Comment:
```suggestion
    errdefer bin_file.destroy();
```

keep `link.File.Elf.deinit` the same. add new function, `destroy()` which calls `deinit()` and then frees the allocation.